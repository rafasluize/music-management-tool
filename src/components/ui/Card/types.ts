export interface ICard {
  children: React.ReactNode;
  title: string;
  sideHeader?: string | React.ReactNode;
}
