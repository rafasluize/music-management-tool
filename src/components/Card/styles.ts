import styled from 'styled-components';

const Wrapper = styled.section`
  background-color: ${({ theme }) => theme.color.background.neutral.default};
  border-radius: ${({ theme }) => theme.border.radius.medium};
  box-shadow: ${({ theme }) => theme.elevation.ton5};
`;

const Header = styled.div`
  border-bottom: ${({ theme }) => theme.border.width.thinner} solid
    ${({ theme }) => theme.color.background.neutral.strong};
  padding: ${({ theme }) => theme.spacing.km4};
  display: flex;
  align-items: center;
  justify-content: space-between;

  color: ${({ theme }) => theme.color.text.secondary.default};
  font-weight: 700;
  font-size: ${({ theme }) => theme.font.size.m6};

  div {
    font-weight: 700;
    font-size: ${({ theme }) => theme.font.size.m2};
    color: ${({ theme }) => theme.color.text.neutral.strong};
  }
`;

const Content = styled.div`
  padding: ${({ theme }) => theme.spacing.km4};
`;

export { Wrapper, Header, Content };
