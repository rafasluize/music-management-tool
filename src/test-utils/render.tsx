import React, { FC, JSXElementConstructor, ReactElement, ReactNode, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { render, RenderOptions } from '@testing-library/react';
import theme from 'styles/theme';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router-dom';

const ThemeDS: FC<{ children: ReactNode }> = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export const wrapperRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) => render(component, { wrapper: ThemeDS, ...options });

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const queryRender = (
  component: ReactElement<Element, string | JSXElementConstructor<Element>>,
  defaultValues?: unknown,
  options?: RenderOptions<typeof import('@testing-library/dom/types/queries'), HTMLElement>,
) =>
  render(
    <Suspense fallback={<p>Carregando</p>}>
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>{component}</BrowserRouter>
      </QueryClientProvider>
    </Suspense>,
    { wrapper: ThemeDS, ...options },
  );

export { wrapperRender as render, queryRender };

export default wrapperRender;
