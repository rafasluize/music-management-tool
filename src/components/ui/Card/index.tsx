import React, { FC } from 'react';

import * as Styled from './styles';
import { ICard } from './types';

const Card: FC<ICard> = ({ children, title, sideHeader }) => {
  return (
    <Styled.Wrapper>
      <Styled.Header>
        {title}

        {sideHeader && <div>{sideHeader}</div>}
      </Styled.Header>

      <Styled.Content>{children}</Styled.Content>
    </Styled.Wrapper>
  );
};

export default Card;
