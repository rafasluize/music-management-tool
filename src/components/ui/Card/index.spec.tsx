import React from 'react';
import { render } from 'test-utils/render';
import Card from '.';

describe('card', () => {
  it('snapshot', () => {
    const { container } = render(<Card title="title here">text</Card>);

    // eslint-disable-next-line testing-library/no-node-access
    const wrapper = container.firstElementChild;

    expect(wrapper).toMatchSnapshot();
  });

  it('title render', () => {
    const { getByText } = render(<Card title="title here">text</Card>);

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const title = getByText('title here');

    expect(title).toBeInTheDocument();
  });

  it('sideHeader render', () => {
    const { getByText } = render(
      <Card title="title here" sideHeader="text here">
        text
      </Card>,
    );

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const title = getByText('text here');

    expect(title).toBeInTheDocument();
  });
});
