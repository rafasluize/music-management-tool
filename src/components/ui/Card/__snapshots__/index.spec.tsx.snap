// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 6px 14px -6px rgba(38, 47, 74, 0.12),0px 10px 32px -4px rgba(24, 38, 47, 0.1);
}

.c1 {
  border-bottom: 1px solid #dddfe4;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: #6c2c83;
  font-weight: 700;
  font-size: 1.25rem;
}

.c1 div {
  font-weight: 700;
  font-size: 0.75rem;
  color: #111111;
  line-height: 2.5;
}

.c2 {
  padding: 16px;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    title here
  </div>
  <div
    class="c2"
  >
    text
  </div>
</section>
`;
